src/aws/mod.rs

```diff
@@ -820,4 +821,54 @@ mod tests {
             store.delete(location).await.unwrap();
         }
     }
+
+    /// Integration test that ensures I/O is done on an alternate threadpool
+    /// when using the `SpawnedReqwestConnector`.
+    #[test]
```

Comment:
I also wrote this end to end integration test to ensure that everything worked correctly when hooked up to an ObjectStore implementation (and it does!)